jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/builders/TravelerMessageFromHumanToAsnConverter.java

```diff
@@ -38,1082 +38,1265 @@
-            && (transformedLonabs & (-1 << 11)) == 0
-            || (transformedLon < 0 && ((transformedLonabs ^ (1 << 11)) == 0))) {
-         // 11 bit value
-         return "node-LL1";
-      } else if (((transformedLatabs & (-1 << 13)) == 0 || (transformedLat < 0 && (transformedLatabs ^ (1 << 13)) == 0))
-            && (transformedLonabs & (-1 << 13)) == 0
-            || (transformedLon < 0 && ((transformedLonabs ^ (1 << 13)) == 0))) {
-         // 13 bit value
-         return "node-LL2";
-      } else if (((transformedLatabs & (-1 << 15)) == 0 || (transformedLat < 0 && (transformedLatabs ^ (1 << 15)) == 0))
-            && (transformedLonabs & (-1 << 15)) == 0
-            || (transformedLon < 0 && ((transformedLonabs ^ (1 << 15)) == 0))) {
-         // 15 bit value
-         return "node-LL3";
-      } else if (((transformedLatabs & (-1 << 17)) == 0 || (transformedLat < 0 && (transformedLatabs ^ (1 << 17)) == 0))
-            && (transformedLonabs & (-1 << 17)) == 0
-            || (transformedLon < 0 && ((transformedLonabs ^ (1 << 17)) == 0))) {
-         // 17 bit value
-         return "node-LL4";
-      } else if (((transformedLatabs & (-1 << 21)) == 0 || (transformedLat < 0 && (transformedLatabs ^ (1 << 21)) == 0))
-            && (transformedLonabs & (-1 << 21)) == 0
-            || (transformedLon < 0 && ((transformedLonabs ^ (1 << 21)) == 0))) {
-         // 21 bit value
-         return "node-LL5";
-      } else if (((transformedLatabs & (-1 << 23)) == 0 || (transformedLat < 0 && (transformedLatabs ^ (1 << 23)) == 0))
-            && (transformedLonabs & (-1 << 23)) == 0
-            || (transformedLon < 0 && ((transformedLonabs ^ (1 << 23)) == 0))) {
-         // 23 bit value
-         return "node-LL6";
+    }
+
+    JsonNode sequence = JsonUtils.newNode().set(SEQUENCE_STRING, newItems);
+
+    dataFrame.remove("items");
+
+    // step 2, set the content CHOICE
+    String replacedContentName = dataFrame.get("content").asText();
+    if (replacedContentName.equals("Advisory")) {
+      replacedContentName = "advisory";
+    }
+
+    // The following field is called "content" but this results in a
+    // failed conversion to XML
+    // see @us.dot.its.jpo.ode.traveler.TimController.publish
+    dataFrame.set(TCONTENT_STRING, JsonUtils.newNode().set(replacedContentName, sequence));
+    dataFrame.remove("content");
+  }
+
+  /**
+   * Builds an item.
+   *
+   * @param itemStr String representing the item
+   * @return JsonNode representing the item
+   */
+  public static JsonNode buildItem(String itemStr) {
+    JsonNode item = null;
+    // check to see if it is a itis code or text
+    try {
+      item = JsonUtils.newNode().set(ITEM, JsonUtils.newNode().put(ITIS, Integer.valueOf(itemStr)));
+      // it's a number, so create "itis" code
+    } catch (NumberFormatException e) {
+      // it's a number, so create "text"

```

Comment:
(non-blocking) Should this comment be updated to say `// it's a string, so create "text"`?

Reply:
Good catch, I've updated this to say `it's not a number, so create text`